framework/src/meshgenerators/AnnularMeshGenerator.C

```diff
@@ -292,7 +300,22 @@ AnnularMeshGenerator::generate()
     }
   }
 
-  mesh->prepare_for_use();
+  // apply boundary id offset and name prefix
+  const auto mesh_boundary_ids = boundary_info.get_boundary_ids();
+  for (auto rit = mesh_boundary_ids.rbegin(); rit != mesh_boundary_ids.rend(); ++rit)
+  {
+
+    const std::string old_sideset_name = boundary_info.sideset_name(*rit);
+    const std::string old_nodeset_name = boundary_info.nodeset_name(*rit);
+
+    MeshTools::Modification::change_boundary_id(*mesh, *rit, *rit + _boundary_id_offset);
 
+    boundary_info.sideset_name(*rit + _boundary_id_offset) =
```

Comment:
same with .empty() on the string prefix

Reply:
i do not understand your suggestion.

Reply:
Don’t change the name of the user did not pass a prefix

Reply:
Added it